marshmallow/marshalling.py

```diff
@@ -230,12 +232,10 @@ def deserialize(
                     data=ret,
                 )
             return ret
-        partial_is_collection = is_collection(partial)
-        ret = dict_class()
-        # Check data is a dict
         if not isinstance(data, collections.Mapping):
             self.store_error(SCHEMA, ('Invalid input type.', ), index=index)
```

Comment:
The error for non-iterables is logged here. I don't like the flow of code now - it's somewhat confusing - but I didn't want to nest the code to leave minimal diff.

Reply:
Oh, and I moved `partial_is_collection` slightly, because it's not needed on the main level. I can restore original code if you want.

Reply:
> The error for non-iterables is logged here. I don't like the flow of code now - it's somewhat confusing - but I didn't want to nest the code to leave minimal diff.

Fair enough. Feel free to make further improvements to the flow here in a future PR.